main.go

```diff
@@ -61,8 +61,9 @@ var (
 	_                     = flag.Bool("write_secrets", false, "[unused]")
 	smConnectionPoolSize  = flag.Int("sm_connection_pool_size", 5, "size of the connection pool for the secret manager API client")
 	iamConnectionPoolSize = flag.Int("iam_connection_pool_size", 5, "size of the connection pool for the IAM API client")
-
-	version = "dev"
+	QPS                   = float32(20.0)
```

Comment:
should we take this from the deployment config ?
same for other

Reply:
Accepting QPS and Burst values from the deployment configuration seems like a good approach, as it offers more flexibility for updates when necessary. There are two options I’m considering for retrieving these values from the config:
- Passing QPS and Burst as environment variables in the DaemonSet at `deploy/provider-gcp-plugin.yaml`.
- Specifying QPS and Burst as parameters within the `SecretProviderClass` for the GCP provider.

Please let me know which approach you think is preferable.

Reply:
`deploy/provider-gcp-plugin.yaml` seems to be a good option. Also I think default behaviour should be the existing values (default values).
@amitmodak WDYT?

Reply:
Updated to configure QPS and Burst from values provided in environment variable in deployment config.